.gitignore

```diff
@@ -21,3 +21,4 @@ docs/src
 pyrightconfig.json
 venv/
 pytest-report.xml
+.python-version
```

Comment:
This is not an artifact produced by the project so I will not incorporate it into the project gitignore, users should use their local global gitignore for that.

Reply:
I see your philosophy here but can we trust all contributors to do that? Is there any harm in simply adding this to the project `.gitignore`?

Reply:
Each contributor should use a global .gitignore for personal files, keeping the project’s .gitignore focused on actual artifacts. Adding one exception sets a precedent for others, leading to unnecessary bloat. Git provides a global .gitignore, precisely for this reason. Just because it’s not immediately harmful doesn’t mean it’s the right approach.

Reply:
I see your point, but we don't seem to have any official project guidance around this (unless I am missing something) and I'm seeing plenty of other artifacts listed in this `.gitignore` related to non-project artifacts - I'd be happy to do a follow-up PR to add that guidance and clean that up, but something like that seems out-of-scope for this PR. For this, I would rather focus on getting rid of `.python-version` and preventing it from getting added back.

Reply:
When a project is young, "official guidance" is whatever the maintainers agree on until the CONTRIBUTING guide is updated. I’ve made my case, so let’s wait for their input. If cleanup is needed, that should be a separate discussion - not a reason to add more non-project artifacts.